manualsource/src/main/java/org/hibernate/search/manualsource/WorkLoad.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+
+package org.hibernate.search.manualsource;
+
+import java.io.Serializable;
+
+/**
+ * Represents an indexing and querying work load.
```

Comment:
I'm not sure I like the mix of indexing and querying in a single contract. Specifically, what are semantics when running queries during an un-finished batch? On first thought, I'd split the two up into two things, e.g. `IndexingTask` and `Query` (yes, I know that name is taken).